packages/pwa/app/components/header/index.test.js

```diff
@@ -29,96 +17,16 @@ jest.mock('@chakra-ui/react', () => {
     }
 })
 
-jest.mock('../../commerce-api/utils', () => {
-    const originalModule = jest.requireActual('../../commerce-api/utils')
-    return {
-        ...originalModule,
-        isTokenValid: jest.fn().mockReturnValue(true)
-    }
+jest.mock('../../commerce-api/hooks/useCustomer', () => {
+    return () => ({
+        isRegistered: true
+    })
```

Comment:
I like the idea of simplifying and only leaving unit tests in this file testing the `<Header />` component.

Here we decided to mock the `useCustomer` hook and remove all the tests verifying that the hook returned the correct data.
If we remove those tests, how will we detect that the hook is working as expected?
Are we planning to add the removed tests to the `useCustomer` hook as part of this PR?

Reply:
I think there is value in having integration tests that test out how different parts of our code work together but I also think we should make a distinction and separate out the unit tests and integration tests. At the moment we try to do both in the same test files and it makes it more complicated to investigate tests when they break.

For useCustomer and other hooks, we should probably have separate tests that focuses on them. At the moment in many of our existing tests, they go through the same paths through the hooks and we end up repeating tests. One example is our account page test and reset-password component test both test the password reset form and the same calls to the useCustomer hook.

When we separate `commerce-api` from the Retail React App, I think our unit tests should mock the hooks while our integration tests can use `msw`.

Reply:
> If we remove those tests and mock the useCustomer hook, how will we detect that the hook is working as expected? How we will find if the component behaves as expected in a more real request-response scenario at the API layer?

Here we are talking about the unit test for the `Header` component, it's not the job of this component unit test to make sure the `useCustomer` hook is functioning. That's the responsibility of the hook's test! A component unit test should solely test the component, and mock every other dependencies. Unit tests should have very specific scope, so we don't end up writing giant unit tests that cover the entire application.

How do we know if a piece of code is well tested? The answer is: check if the code has unit test!

In this case, how do we know if `useCustomer` hook is tested? well, the answer is definitely not "Check if there are header component unit tests", the answer is "Check if there are useCustomer tests".

Reply:
There are many layers that we could mock:

```
react component
         |
Commerce API hooks
         |
Commerce API isomorphic SDK
         |
HTTP response
```

Today, in many of our component unit tests, we are mocking the HTTP responses, and the tests cover the top 3 layers. 
The test itself have to understand the hook implementation and setup the states of the hooks. This makes the tests way too complicated for component unit testing. It's unnecessary knowledge to the tests.

A BAD example inside component unit test:
```jsx
const MockedComponent = (props) => {
    const customer = useCustomer()

    useEffect(() => {
        if (!customer.isRegistered) {
            customer.login('customer@test.com', 'password1')
        }
    }, [])

    return (
        <div>
            <div>firstName: {customer?.firstName}</div>
            <ButtonWithRegistration {...props}>Button</ButtonWithRegistration>
        </div>
    )
}
```

In this bad example, the unit test of a component calls the hook method `login`, this leaks too much implementation detail from the hook into the unit tests. Imagine we want to extract the components to a different package, `customer.login()` will make no sense in the new package. This is a red flag for unit tests.

The correct way to mock would be:
```
jest.mock('../../commerce-api/hooks/useCustomer', () => {isRegistered: true})
```

Why is the second example correct? because the component implementation access the `isRegistered` property, like `customer.isRegistered`, and we want to test the implementation of the component (rather than the implementation of the `useCustomer` hook, e.g. it set `isRegistered` state from the `login` method)

Reply:
> How do we know if a piece of code is well tested? The answer is: check if the code has unit test!

> In this case, how do we know if useCustomer hook is tested? well, the answer is definitely not "Check if there are header component unit tests", the answer is "Check if there are useCustomer tests".

Hence the comment. The `useCustomer` hook doesn’t have unit tests.